modules/aggregation-layers/src/utils/gpu-grid-aggregation/gpu-grid-aggregator.js

```diff
@@ -201,6 +214,46 @@ export default class GPUGridAggregator {
     return this.runAggregationOnCPU(aggregationParams);
   }
 
+  // Reads aggregation data into JS Array object
+  // For WebGL1, data is available in JS Array objects already.
+  // For WebGL2, data is read from Buffer objects and cached for subsequent queries.
+  /* eslint-disable max-depth */
+  getData(weights = null) {
```

Comment:
Personally I dislike APIs that return nested results. The gained convenience is trivial compared to the added complexity and loss of readability. Compare your code:

```js
const aggregationResults = gpuGridAggregator.getData('color');
...
Object.assign({pixelIndex: index}, aggregationResults.color)
```

vs

```js
const aggregationResults = {
  color: gpuGridAggregator.getData('color')
};
...
Object.assign({pixelIndex: index}, aggregationResults.color)
```

The first one contains an implicit data structure that is difficult to track down.

Reply:
Intention was to provide results for more than one weight with a single `getData` call.  If preferred, I can make argument to `getData` non optional and remove nesting in return object.

Reply:
Removed nesting.